buildkite/build_release.sh

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env bash
+
+set -xueo pipefail
+trap 'echo build script failed at line $LINENO' ERR
+
+TODAY=$(date +'%Y-%m-%d')
+: ${BRANCH:=master}
+
+if [ ! -d installer ]; then
+    git clone https://github.com/dlang/installer.git installer
+fi
+pushd installer/create_dmd_release
+
+git reset --hard --quiet
+git clean -df --quiet
+if [ $BRANCH != master ] && [ $BRANCH != stable ] &&
+       ! git ls-remote --exit-code --heads origin $BRANCH > /dev/null; then
+    # use master as fallback base for feature branches
+    git fetch origin master --quiet
+else
+    git fetch origin $BRANCH --quiet
+fi
+git checkout FETCH_HEAD --quiet
+
+mkdir build
+LATEST=$(curl -fsSL http://downloads.dlang.org/releases/LATEST)
+rdmd build_all.d v$LATEST $BRANCH && success=yes || success=no
+rm -f build/*.zip # only keep lzma archives
+
+for file in build/*; do
+    gpg --detach-sign --no-use-agent $file
+done
+
+mv build dmd-$BRANCH-$TODAY
+aws --profile ddo s3 sync dmd-$BRANCH-$TODAY/ s3://downloads.dlang.org/nightlies/dmd-$BRANCH-$TODAY/ --acl public-read --cache-control max-age=604800
+popd
+
+if [ ! -d downloads.dlang.org ]; then
+    git clone https://github.com/dlang/downloads.dlang.org.git downloads.dlang.org
+else
+    git -C downloads.dlang.org pull
+fi
+pushd downloads.dlang.org
+if [ ! -f config.json ]; then
+    (umask 0277 && cat > config.json) <<EOS
+{
+    "aws": {
+        "key": "$(sed -n 's|aws_access_key_id\s*=\s*||p' ~/.aws/credentials)",
+        "secret": "$(sed -n 's|aws_secret_access_key\s*=\s*||p' ~/.aws/credentials)",
+        "endpoint": "s3-$(sed -n 's|region\s*=\s||p' ~/.aws/config).amazonaws.com"
+    },
+    "s3bucket": "downloads.dlang.org"
+}
+EOS
+fi
+
+# create redirects from dmd-master/* to dmd-master-2018-10-12/*
+if [ $success = yes ]; then
+    for file in $(ls ../installer/create_dmd_release/dmd-$BRANCH-$TODAY); do
+        echo -n | aws --profile ddo s3 cp - s3://downloads.dlang.org/nightlies/dmd-$BRANCH/$file --acl public-read --cache-control max-age=604800 --website-redirect http://downloads.dlang.org/nightlies/dmd-$BRANCH-$TODAY/$file
+    done
+    echo -n | aws --profile ddo s3 cp - s3://downloads.dlang.org/nightlies/dmd-$BRANCH --acl public-read --cache-control max-age=604800 --website-redirect http://downloads.dlang.org/nightlies/dmd-$BRANCH-$TODAY
+    echo -n $BRANCH-$TODAY | aws --profile ddo s3 cp - s3://downloads.dlang.org/nightlies/dmd-$BRANCH/LATEST --acl public-read --cache-control max-age=604800
+fi
+
+make -C src
+rm -rf ddo
+./src/build-gen-index s3_index generate
+aws --profile ddo s3 sync ./ddo/nightlies/ s3://downloads.dlang.org/nightlies/ --acl public-read --cache-control max-age=604800
```

Comment:
Maybe sync the files _first_ before you do the redirect?
(maybe the sync breaks and or the internet becomes slow ...)

Reply:
That's just syncing the index.html files, the actual binaries are uploaded beforehand.